File: sort/insertionSort.py
#! /usr/bin/env python 
from util import *


def reverseSort(arr, size):
    i = size

    while (i >= 0):
        if arr[i] > arr[i + 1]:
            swap(arr, i ,i + 1)
        i -= 1

def insertionSort(arr):
    for i in range(len(arr) - 1):
        if arr[i] > arr[i + 1]:
            reverseSort(arr, i)
    
    return arr

File: sort/insertionSort.c
#include "util.h"
#include "insertionSort.h"


void insertionSort(int *arr, int size)
{
    for (int i = 0; i < size - 1; i++)
    {
        if (arr[i] > arr[i + 1])
            reverseSort(arr, i);
    }
}


void reverseSort(int* arr, int size)
{
    for (int i = size; i >= 0; i--)
    {
        if (arr[i] > arr[i + 1])
            swap(arr, i, i + 1);
    }
}